### DISCUSSION TITLE
Client side Authorisation Module

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#359] [Ideas]
I have written a module as an add on to the Authorisation module that takes care of hiding/showing and enabling/disabling components based on the user's permissions.

It is quite simple but I have been using it for the last month or so and it seems to be working very well. Obviously it needs to be used in conjunction with Server side authentication (which the current module does already) but it does make it very easy to build a form for all users and then hide components dynamically.

Would this be a valuable addition to the project? If so I am happy to create a demo app for some further feedback?

**Comment by meatballs** [comment]
If a demo would be quick to put together, go for it!

**Comment by lewysigns** [comment]
This would definitely be useful imo! I already have a hacked up client side authorization set up as well, but I would much prefer it be part of the anvil_extras library. ( And definitely better than what I have written.)

**Comment by rhurlbatt** [comment]
Ok so I have thrown together a quick demo with basic usage - apologies for the lack of bells and whistles.

[Example Link](https://plaintive-outstanding-object.anvil.app)

[Clone Link](https://anvil.works/build#clone:Z3PBYMQAAHUDGB5E=FJ5XCP4LCVGUT2LXGV2U73XK)

It basically has two user profiles set up.
The usernames are 'adminuser' and 'basicuser'
Passwords are both 'password'

It works well with routing but I am sure it can be optimised quite a bit. 

Obviously it needs the server side authorisation module to make it secure.

I think there is still a fair amount to work out (ie how to deal with user changes) but if there is some interest in adding this to Anvil Extras I will happily work out what is missing for a more well rounded module.

**Comment by meatballs** [comment]
My apologies, but I'm not likely to get time to look at this properly. (I'm in the middle of trying to move house and also organise the PyCon UK schedule at the moment).
